Pokepedia.ApiAdapter/PokeApi/GetLocation.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Newtonsoft.Json;
+using Pokepedia.ApiAdapter.Models;
+using System.Net.Http.Headers;
+
+namespace Pokepedia.ApiAdapter.PokeApi
+{
+    public class GetLocation
+    {
+        private const string Url = "https://pokeapi.co/api/v2/";
+
+        public static async Task<LocationModel> GetLocationByPokemonNameAsync(string name)
+        {
+            var urlQuery = $"pokemon/{name}/encounters";
+            return await GetLocationAsync(urlQuery);
+        }
+
+        private static async Task<LocationModel> GetLocationAsync(string urlQuery)
```

Comment:
This same method is the same for getlocation and getpokemon right? So we have duplicated code. You agree? I wonder how we could share this method across those two files? HMMMMMM (do it).. Maybe a new file that had this shared method shrug

Reply:
the basic answer seems like putting the method into a static class and calling it, but the issue is that they both use different tasks, Is that really duplicated code